src/StochasticNoiseSimulator.cpp

```diff
@@ -269,13 +305,27 @@ void StochasticNoiseSimulator::runStochSimulationForId(unsigned int
         localDD->decRef(localRootEdge);
         const auto t2 = std::chrono::steady_clock::now();
 
+        //        for (bool classic_value : classic_values) {
+        std::string classic_register_string;
+        if (!classic_values.empty()) {
+            for (unsigned long i = 0; i < classic_values.size(); i++) {
+                if (classic_values[i]) {
+                    classic_register_string.push_back('0');
+                } else {
+                    classic_register_string.push_back('1');
+                }
+            }
```

Comment:
use range-based for loop
```suggestion
            for (const auto val: classicalValues)
                classicalRegisterString.push_back(val? '0': '1');
```

Reply:
done